### PR TITLE
executed the task

### PR DESCRIPTION
**Pull request by SerhiiAN**

### CODE REVIEW

File: src/main/java/mate/academy/intro/model/Book.java [hunk 1]
```diff
@@ -27,5 +31,8 @@ public class Book {
     @Column(nullable = false)
     private BigDecimal price;
     private String description;
+    @Column(name = "cover_image")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/intro/model/Book.java]:
```suggestion
```

File: src/main/java/mate/academy/intro/controller/BookController.java [hunk 2]
```diff
@@ -32,4 +35,10 @@ public BookDto getBookById(@PathVariable Long id) {
     public BookDto createBook(@RequestBody CreateBookRequestDto requestDto) {
         return bookService.save(requestDto);
     }
+
+    @PutMapping("/{id}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void updateById(@PathVariable Long id) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/intro/controller/BookController.java]: Pass the request DTO, 
Screenshot 2024-06-21 at 08 59 33

File: src/main/java/mate/academy/intro/service/BookServiceImpl.java [hunk 3]
```diff
@@ -30,9 +30,19 @@ public List<BookDto> findAll() {
...
         );
         return bookMapper.toDto(book);
     }
+
+    @Override
+    public BookDto updateById(Long id) {
+        Book book = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can not find book " + id)
+                );
+        book.setDeleted(true);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/intro/service/BookServiceImpl.java]:
```suggestion
       bookMapper.updateBookFromDto(bookDto, book);

```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job, let's fix some comments

File: pom.xml [hunk 4]
```diff
@@ -58,6 +58,11 @@
             <artifactId>mapstruct</artifactId>
             <version>${mapstruct.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.liquibase</groupId>
+            <artifactId>liquibase-core</artifactId>
+            <version>4.27.0</version>
```
Comment (Elena-Bruyako) [inline pom.xml]:
```suggestion
            <version>${liquibase.version}</version>
```

spring-boot-parent provides this version

File: pom.xml [hunk 5]
```diff
@@ -110,6 +115,14 @@
                     </annotationProcessorPaths>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.liquibase</groupId>
+                <artifactId>liquibase-maven-plugin</artifactId>
+                <version>4.27.0</version>
```
Comment (Elena-Bruyako) [inline pom.xml]:
```suggestion
                <version>${liquibase.version}</version>
```

File: src/main/java/mate/academy/intro/model/Book.java [hunk 6]
```diff
@@ -28,4 +32,6 @@ public class Book {
     private BigDecimal price;
     private String description;
     private String coverImage;
+    @Column(name = "is_deleted", nullable = false, columnDefinition = "TINYINT(1) DEFAULT 0")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/intro/model/Book.java]:
```suggestion
    @Column(nullable = false, columnDefinition = "TINYINT(1) DEFAULT 0")
```

Hibernate automatically create column in snake_case style

File: src/main/resources/application.properties [hunk 7]
```diff
@@ -1,8 +1,9 @@
...
 spring.datasource.username=root
 spring.datasource.password=1234
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
-
-spring.jpa.hibernate.ddl-auto=create-drop
+spring.jpa.hibernate.ddl-auto=validate
 spring.jpa.show-sql=true
+spring.jpa.open-in-view=false
+
+
```
Comment (Elena-Bruyako) [inline src/main/resources/application.properties]:
```suggestion
```

File: src/main/resources/db/changelog/changes/02-change-column-cover_image-in-table-books.yaml [hunk 8]
```diff
@@ -0,0 +1,11 @@
+databaseChangeLog:
+  - changeSet:
+      id: rename-column-cover_image-to-coverImage
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/02-change-column-cover_image-in-table-books.yaml]: remove this changelog

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Need to fix some comments

File: src/main/java/mate/academy/intro/model/Book.java [hunk 9]
```diff
@@ -9,8 +9,12 @@
 import java.math.BigDecimal;
 import lombok.Getter;
 import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
 
 @Entity
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id=?")
+@SQLRestriction(value = "is_deleted=false")
```
Comment (okuzan) [inline src/main/java/mate/academy/intro/model/Book.java:17]:
```suggestion
@SQLDelete(sql = "UPDATE books SET is_deleted = TRUE WHERE id = ?")
@SQLRestriction(value = "is_deleted = FALSE")
```

File: src/main/java/mate/academy/intro/service/BookServiceImpl.java [hunk 10]
```diff
@@ -30,9 +30,24 @@ public List<BookDto> findAll() {
...
                 () -> new EntityNotFoundException("Cannot find Book with id: " + id)
         );
         return bookMapper.toDto(book);
     }
+
+    @Override
+    public BookDto updateById(Long id, CreateBookRequestDto requestDto) {
+        Book book = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can not find book " + id)
+                );
```
Comment (okuzan) [inline src/main/java/mate/academy/intro/service/BookServiceImpl.java:43]:
```suggestion
                () -> new EntityNotFoundException("Can not find book " + id));
```

**Review by okuzan — APPROVED** [review]

**Pull request merged**